pkg/controllers/workgenerator/controller.go

```diff
@@ -135,18 +136,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, req controllerruntime.Reques
 
 	workUpdated := false
 	overrideSucceeded := false
-	// Reset the conditions and failed placements.
-	for i := condition.OverriddenCondition; i < condition.TotalCondition; i++ {
-		resourceBinding.RemoveCondition(string(i.ResourceBindingConditionType()))
-	}
 	resourceBinding.Status.FailedPlacements = nil
 	// list all the corresponding works
 	works, syncErr := r.listAllWorksAssociated(ctx, &resourceBinding)
 	if syncErr == nil {
 		// generate and apply the workUpdated works if we have all the works
 		overrideSucceeded, workUpdated, syncErr = r.syncAllWork(ctx, &resourceBinding, works, cluster)
 	}
-
+	// Reset the conditions and failed placements.
```

Comment:
Hi Ryan! Just a nit: is line 142 (139) `resourceBinding.Status.FailedPlacements = nil` being left behind?